src/gps/gpsTask.c

```diff
@@ -29,18 +29,21 @@ void GPSTask(void *pvParameters)
     Serial *serial = get_serial(SERIAL_GPS);
     uint8_t targetSampleRate = decodeSampleRate(getWorkingLoggerConfig()->GPSConfigs.speed.sampleRate);
     lapStats_init();
-    gps_status_t gps_status = GPS_init(targetSampleRate, serial);
-    if (!gps_status) {
-        pr_error("GPS: Error provisioning\r\n");
-    }
+    while(1) {
+        gps_status_t gps_status = GPS_init(targetSampleRate, serial);
+        if (!gps_status) {
+            pr_error("GPS: Error provisioning\r\n");
```

Comment:
It seems that GPS message mis alignment causes a problem with how we
process it with dma. This is like a watchdog for the GPS module if
something hiccups in their firmware, too.

Brent Picasso
Autosport Labs
Technology for Race and Street
On May 19, 2015 5:19 PM, "Andrew Stiegmann" <notifications@github.com>
wrote:

> In src/gps/gpsTask.c
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/49#discussion_r30661723>
> :
>
> > @@ -29,18 +29,21 @@ void GPSTask(void *pvParameters)
> >      Serial *serial = get_serial(SERIAL_GPS);
> >      uint8_t targetSampleRate = decodeSampleRate(getWorkingLoggerConfig()->GPSConfigs.speed.sampleRate);
> >      lapStats_init();
> > -    gps_status_t gps_status = GPS_init(targetSampleRate, serial);
> > -    if (!gps_status) {
> > -        pr_error("GPS: Error provisioning\r\n");
> > -    }
> > +    while(1) {
> > +        gps_status_t gps_status = GPS_init(targetSampleRate, serial);
> > +        if (!gps_status) {
> > +            pr_error("GPS: Error provisioning\r\n");
>
> So if we fail to provision do we need to backoff? Why would it work the
> second time and not the first?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/autosportlabs/RaceCapture-Pro_firmware/pull/49/files#r30661723>
> .
>